src/Storages/StorageReplicatedMergeTree.cpp

```diff
@@ -7998,10 +7998,19 @@ void StorageReplicatedMergeTree::replacePartitionFrom(
             replace = false;
         }
 
+        scope_guard intent_guard;
         if (!replace)
         {
             /// It's ATTACH PARTITION FROM, not REPLACE PARTITION. We have to reset drop range
             drop_range = makeDummyDropRangeForMovePartitionOrAttachPartitionFrom(partition_id);
+            queue.addDropReplaceIntent(drop_range);
+            intent_guard = scope_guard{[this, my_drop_range = drop_range]() { queue.removeDropReplaceIntent(my_drop_range); }};
+
+            getContext()->getMergeList().cancelInPartition(getStorageID(), drop_range.partition_id, drop_range.max_block);
+            {
+                auto pause_checking_parts = part_check_thread.pausePartsCheck();
+                part_check_thread.cancelRemovedPartsCheck(drop_range);
+            }
```

Comment:
AFAIU it's not necessary here, there's no real DROP/REPLACE_RANGE

Reply:
But we need it outside of `if (!replace)`

Reply:
We only need it when `replace == true`, right? Otherwise we don't remove parts, so new parts can appear without any issues in my understanding.

Reply:
Yep